etcd-manager/pkg/volumes/scaleway/volumes.go

```diff
@@ -0,0 +1,248 @@
+				localEtcdVolume.LocalDevice = fmt.Sprintf("%s%s", localDevicePrefix, volume.ID)
+			}
+		}
+
+		localEtcdVolumes = append(localEtcdVolumes, localEtcdVolume)
+	}
+
+	return localEtcdVolumes, nil
+}
+
+// FindMountedVolume returns the device where the volume is mounted to the running server.
+func (a *ScwVolumes) FindMountedVolume(volume *volumes.Volume) (string, error) {
+	device := volume.LocalDevice
+
+	klog.V(2).Infof("Finding mounted volume %q", device)
+	_, err := os.Stat(volumes.PathFor(device))
+	if err == nil {
+		klog.V(2).Infof("Found mounted volume %q", device)
+		return device, nil
+	}
+
+	if !os.IsNotExist(err) {
+		return "", fmt.Errorf("failed to find local device %q: %v", device, err)
+	}
+
+	// When not found, the interface says to return ("", nil)
+	return "", nil
+}
+
+// AttachVolume attaches the specified volume to the running server and returns the mountpoint if successful.
+func (a *ScwVolumes) AttachVolume(volume *volumes.Volume) error {
+	for {
+		volumeResp, err := a.instanceAPI.GetVolume(&instance.GetVolumeRequest{
+			VolumeID: volume.ProviderID,
+			Zone:     a.zone,
+		})
+		if err != nil || volumeResp.Volume == nil {
+			return fmt.Errorf("failed to get info for volume id %q: %w", volume.ProviderID, err)
+		}
+
+		scwVolume := volumeResp.Volume
+		if scwVolume.Server != nil {
+			if scwVolume.Server.ID != a.server.ID {
+				return fmt.Errorf("found volume %s(%s) attached to a different server: %s", scwVolume.Name, scwVolume.ID, scwVolume.Server.ID)
+			}
+			klog.V(2).Infof("Attached volume %s(%d) to the running server", scwVolume.Name, scwVolume.ID)
+			volume.LocalDevice = fmt.Sprintf("%s%s", localDevicePrefix, scwVolume.ID)
+			return nil
+		}
+
+		//err = a.instanceAPI.ServerActionAndWait(&instance.ServerActionAndWaitRequest{
+		//	ServerID: a.server.ID,
+		//	Zone:     a.zone,
+		//	Action:   instance.ServerActionPoweroff,
+		//})
+		//if err != nil {
+		//	return fmt.Errorf("failed to power-off server %s before attaching volume", a.server.ID)
+		//}
+
+		klog.V(2).Infof("Attaching volume %s(%d) to the running server", scwVolume.Name, scwVolume.ID)
+		_, err = a.instanceAPI.AttachVolume(&instance.AttachVolumeRequest{
+			Zone:     a.zone,
+			ServerID: a.server.ID,
+			VolumeID: scwVolume.ID,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to attach volume %s(%s): %w", scwVolume.Name, scwVolume.ID, err)
+		}
+		//if vol.Server.State != instance.ServerStateRunning && vol.Server.State != instance.ServerStateStarting {
+		//	return fmt.Errorf("found invalid status for volume %s(%d): %s", scwVolume.Name, scwVolume.ID, action.Status)
+		//}
+		//_, err = a.instanceAPI.ServerAction(&instance.ServerActionRequest{
+		//	ServerID: a.server.ID,
+		//	Zone:     a.zone,
+		//	Action:   instance.ServerActionPoweron,
+		//})
+		//if err != nil {
+		//	return fmt.Errorf("failed to power server %s back on after attaching volume", a.server.ID)
+		//}
+
+		time.Sleep(10 * time.Second)
```

Comment:
Use a wait operation on this. Having a static wait is not performant nor elegant.